### DISCUSSION TITLE
New bug today? SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.

### DISCUSSION THREAD

**Discussion by EwGadom** [appium/appium#17862] [Q&A]
Hello everyone. Did somebody start the day like me? In every machines (Windows) my framework isn't working...
Yesterday evening it worked properly, I didn't change anything.

Java 17, UiAutomator2.

```
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Host info: {host info}
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:559)
at io.appium.java_client.AppiumDriver.startSession(AppiumDriver.java:229)
at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:157)
at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:80)
at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:92)
at configuration.drivers.Drivers.initializeAndroidDriver(Drivers.java:74)
at basetest.BaseTest.setUp(BaseTest.java:42)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.openqa.selenium.WebDriverException: java.lang.NoSuchMethodException: org.openqa.selenium.remote.ProtocolHandshake.createSession(org.openqa.selenium.remote.http.HttpHandler,java.io.InputStream,long)
Build info: version: '4.7.0', revision: '0a5b49d16f'
System info: os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '17.0.2'
Driver info: driver.version: AppiumDriver
at io.appium.java_client.remote.AppiumProtocolHandshake.createSession(AppiumProtocolHandshake.java:130)
at io.appium.java_client.remote.AppiumProtocolHandshake.createSession(AppiumProtocolHandshake.java:102)
at io.appium.java_client.remote.AppiumCommandExecutor.createSession(AppiumCommandExecutor.java:155)
at io.appium.java_client.remote.AppiumCommandExecutor.execute(AppiumCommandExecutor.java:189)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:541)
... 32 more
Caused by: java.lang.NoSuchMethodException: org.openqa.selenium.remote.ProtocolHandshake.createSession(org.openqa.selenium.remote.http.HttpHandler,java.io.InputStream,long)
at java.base/java.lang.Class.getDeclaredMethod(Class.java:2675)
at io.appium.java_client.remote.AppiumProtocolHandshake.createSession(AppiumProtocolHandshake.java:122)
... 36 more
... Removed 21 stack frames
```

**Comment by jlipps** [comment]
How did you attempt to use the information in the error message to resolve the issue?

**Comment by EwGadom** [comment]
The problem occurred during creating a new instance of AppiumDriver. Helped using ver. 8.3.0 of java-client. It looks like versions 8.0.0 to 8.2.1 are affected but I don't understand why (they work before correctly). For those versions, adding older dependencies  (e.g. 4.5.3.) of  selenium-api, selenium-support, selenium-remote-driver, selenium-chrome-driver than 4.7.0 upgraded in https://github.com/appium/java-client/commit/843941aff92e89080dccd1f704b9201d3a945921 helps. I checked by maven helper in a clean new project including only dependency of java-client that it uses 4.7.0 version of above-mentioned libraries.

**Comment by dsaner** [comment]
I had this exact same issue with the exact same timing.  Upgrading the appium java-client to 8.3.0 fixed the problem.  Not sure what happened to cause 8.2.0 to stop working.  It had been fine for the last month or so.

**Comment by nnson0310** [comment]
As my coaching used to said: "The newest is not always the best". Do not upgrade dependency version at all cost. I am still keeping using Selenium 3.141.59 and Java Client version 7 because they runs stably. Let's try downgrading the version and see if it works.

**Reply by EwGadom** [reply]
Don't you have any problem with the newest Android systems? We work with the newest Android systems still.

**Reply by nnson0310** [reply]
Actually no. I have built an automation source code to run testcases for mobile browser using these old version for a few months and until now it still runs quite stably. I used both emulators and real devices. For emulators, its android version is 9. For real devices (Samsung and Xiaomi), its android version is 12 and 13.  So Android version might not be the root cause here. Try to downgrade version of Selenium and Java Client and run some samples to see how it works. This is the best advice I can give to you right now

**Comment by snoopgr2** [comment]
Seeing this again as of 3/29/2023.  I was using io.appium build 9.2.0 for the past month with no issues.  Suddenly I couldn't work at all.  

After a full day of panic and tears I found this post (thank you so much to those above) and moved back to io.appium 8.6.0 to solve the issue.  (Note I had tried moving back to 9.0.0 but that wasn't far enough so abandoned io.appium as the issue and spent the day reinstalling java and vscode, cursing, weeping, and praying to dark elder gods for deliverance.)

I would love to know why 9.2 suddenly hit this issue when the same code base was working just fine the day before.  My assumption is a maven clean install pulled a new dependency that borked the whole thing but that knowledge is above my pay scale.

**Reply by mykola-mokhnach** [reply]
This happened because Selenium has released a new version, which was incompatible with java client whose version was the recent at that time. See https://github.com/appium/java-client?tab=readme-ov-file#why-is-it-so-complicated

You probably want to pin Selenium version to avoid such issues in the future. Check the [compatibility matrix](https://github.com/appium/java-client?tab=readme-ov-file#compatibility-matrix) for more details

**Reply by preethisharath** [reply]
This happened to me as well 5 days ago. I was in tears too. I moved back to 8.6.0 version. The best thing to avoid such vulnerability could be under eclipse, go to window-> preferences -> Maven and enable Offline option so that this unnecessary update is avoided.

**Comment by sseedarla** [comment]
this was working perfect for me thanks for the help members who posted the solution even Chat Gpt could not suggest :) properly


`

    <dependencies>
        
        <dependency>
            <groupId>io.appium</groupId>
            <artifactId>java-client</artifactId>
            <version>8.6.0</version> 
        </dependency>

        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>4.9.1</version> 
        </dependency>

        
        <dependency>
            <groupId>org.testng</groupId>
            <artifactId>testng</artifactId>
            <version>7.7.1</version>
            <scope>compile</scope>
        </dependency>
        
        <dependency>
            <groupId>org.json</groupId>
            <artifactId>json</artifactId>
            <version>20230227</version>
        </dependency>
        
        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-core</artifactId>
            <version>2.14.1</version>
        </dependency>
        
        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-api</artifactId>
            <version>2.14.1</version>
        </dependency>
        
        <dependency>
            <groupId>com.aventstack</groupId>
            <artifactId>extentreports</artifactId>
            <version>5.0.9</version>
        </dependency>
        
        <dependency>
            <groupId>commons-codec</groupId>
            <artifactId>commons-codec</artifactId>
            <version>1.15</version>
        </dependency>
        
        <dependency>
            <groupId>commons-io</groupId>
            <artifactId>commons-io</artifactId>
            <version>2.11.0</version>
        </dependency>
    </dependencies>


`
